add R scripts for accessing BigQueryDB data tables & cleaning drug_exposure and concept extract from BigQuery DB

zsenousy opened the pull request:
add R script for cleaning drug_exposure extract from BigQuery DB. Description of the issue and detailed description of the steps taken are present in #32.

zsenousy committed:
add R script for cleaning drug_exposure extract

add R script for cleaning drug_exposure extract from BigQuery DB

zsenousy committed:
replace specific paths to generic ones

AngharadGreen commented on R/cleaning_drug_exposure_extract_from_bigQueryDB.R:
```diff
@@ -0,0 +1,32 @@
+library(readr)
+
+# Step 1: Read the drug exposure data
+drug_exposure <- read_delim("./path/to/your/file/uncleaned_drug_exposure.csv", delim = " ")
```
```suggestion
drug_exposure <- read_delim("../data/OMOP/uncleaned_drug_exposure.csv", delim = " ")

```

AngharadGreen commented on R/cleaning_drug_exposure_extract_from_bigQueryDB.R:
```diff
@@ -0,0 +1,32 @@
+drug_exposure$drug_exposure_end_datetime[drug_exposure$drug_exposure_end_datetime == "NA NA"] <- NA
+
+# Step 7: Save the cleaned data to a CSV file
+write_csv(drug_exposure, "./path/to/your/file/cleaned_drug_exposure.csv")
```
```suggestion
write_csv(drug_exposure, "../data/OMOP/cleaned_drug_exposure.csv")

```

AngharadGreen commented on R/cleaning_drug_exposure_extract_from_bigQueryDB.R:
```diff
@@ -0,0 +1,32 @@
+# Step 2: Check the structure of the data
+str(drug_exposure)
+
+# Step 3: Rename columns for clarity
```
A more detailed comment here would be useful explaining why the column names need to be renamed

AngharadGreen commented on R/cleaning_drug_exposure_extract_from_bigQueryDB.R:
```diff
@@ -0,0 +1,32 @@
+# Load necessary library
+library(readr)
```
Should we also include at the beginning the code for how the data was extracted from the BigQuery DB?

zsenousy replied:
A comment will be added to refer to another script loaded for accessing BigQuery DB.

```suggestion
# Please refer to '[Accessing_BigQuery_DB.R](https://github.com/SAFEHR-data/ramses-package/blob/Zsenousy/32-access-and-clean-data-tables-extracted-from-BigQuery-DB/R/Accessing_BigQuery_DB.R) to learn how to generate extract of drug_exposure and concept tables. The resulted data tables will be uncleaned and will require further cleaning as shown here in this script
# Load necessary library
library(readr)

```

AngharadGreen reviewed (changes requested):
Thanks @zsenousy for this I have just a few edits/comments before it can be merged

zsenousy committed:
replace generic path by specific path

Co-authored-by: Angharad <163411123+AngharadGreen@users.noreply.github.com>

zsenousy committed:
replace generic path by specific path

Co-authored-by: Angharad <163411123+AngharadGreen@users.noreply.github.com>

zsenousy committed:
switch step 3 by 4 and add clarify comment on step 4 (rename columns)

zsenousy committed:
add comment to refer reader to 'access_BigQuery_DB' script before cleaning scripts

zsenousy committed:
update comment on the top of the script

zsenousy committed:
add R script for accessing BigQuery DB

add R script for accessing data tables of drug_exposure and concept from BigQuery DB.

zsenousy committed:
cleaning_concept_extract_from_bigQueryDB

cleaning_concept_extract_from_bigQueryDB.R

zsenousy committed:
add comment on top of concept script to guide reader to access BQ DB script

AngharadGreen commented on R/Accessing_BigQuery_DB.R:
```diff
@@ -0,0 +1,43 @@
+library(bigrquery)
+library(DBI)
+library(gargle)
+
+# Step 2: Authenticate connection to BigQuery
```
It may be useful here to include a link https://console.cloud.google.com/ to indicate where a project needs to be set up.

zsenousy replied:
```suggestion
# For instructions on creating a project and generating a key JSON file, 
# please follow instructions: https://github.com/SAFEHR-data/ramses-package/issues/15
# Step 2: Authenticate connection to BigQuery

```

AngharadGreen commented on R/Accessing_BigQuery_DB.R:
```diff
@@ -0,0 +1,43 @@
+# Query to retrieve data from the drug_exposure table, limiting to 10,000 rows
+sql_string1 <- "SELECT * FROM bigquery-public-data.cms_synthetic_patient_data_omop.drug_exposure LIMIT 10000"
+
+# Query to retrieve data from the concept table, filtering by specific domain_ids
```
Is there a reason why just these domain_ids were chosen and not all selected?

zsenousy replied:
I selected only three `domain_id` values—`'Drug'`, `'Route'`, and `'Unit'`—from the `concept` table to focus on the most relevant data for the analysis conducted on drug exposures. @razekmh suggested filtering `concept `on `Drug` domain which gives me information on medications, and I thought about using both the `Route` domain which specifies how these drugs are administered and the `Unit` domain which defines the measurement units for dosages. I thought about avoiding pulling in unnecessary data, making the analysis more efficient and targeted. This way, I can concentrate on the key concepts that directly impact drug exposure. 

I believe you can try experimenting with all the concept table for the sake of exploring some unseen trends.

AngharadGreen commented on R/Accessing_BigQuery_DB.R:
```diff
@@ -0,0 +1,43 @@
+
+# Step 4: Save data to CSV files
+# Save the drug_exposure query result to a CSV file
+write.table(result1, file = "path/to/your/uncleaned_drug_exposure_file.csv", sep = ",", row.names = FALSE, quote = TRUE)
```
Is there a reason write.table was used instead of write.csv? normally write.table produces txt files, not sure if this affects the output csv that is created

zsenousy replied:
I don't think it makes any difference given some arguments are decided such as `row.names = FALSE`. So, `write.table` has more flexibility in dealing with different formats. However,` write_csv` is a straightforward option as well.

AngharadGreen commented on R/Accessing_BigQuery_DB.R:
```diff
@@ -0,0 +1,43 @@
+write.table(result1, file = "path/to/your/uncleaned_drug_exposure_file.csv", sep = ",", row.names = FALSE, quote = TRUE)
+
+# Save the concept query result to a CSV file
+write.table(result2, file = "path/to/your/uncleaned_concept_file.csv", sep = ",", row.names = FALSE, quote = TRUE)
```
Should we also save the uncleaned files in the ../data/OMOP/ folder?

zsenousy replied:
As these file will be generated from the script after successful extract process, I don't recommend adding a resultant file from a script. I support adding files which are used as input to a script. Happy to add them anyway.

AngharadGreen reviewed (changes requested):
Thanks for this @zsenousy, I have made a few comments and suggested edits to be addressed before merging

zsenousy commented on R/Accessing_BigQuery_DB.R:
```diff
@@ -0,0 +1,43 @@
+# Save the drug_exposure query result to a CSV file
+write.table(result1, file = "path/to/your/uncleaned_drug_exposure_file.csv", sep = ",", row.names = FALSE, quote = TRUE)
+
+# Save the concept query result to a CSV file
+write.table(result2, file = "path/to/your/uncleaned_concept_file.csv", sep = ",", row.names = FALSE, quote = TRUE)
```
```suggestion
# Save the concept query result to a CSV file
write.table(result2, file = "../data/OMOP/uncleaned_concept_file.csv", sep = ",", row.names = FALSE, quote = TRUE)

```

zsenousy committed:
replace path to specific one

AngharadGreen reviewed (approved):
Looks good @zsenousy thank you

zsenousy committed:
add reference on instructions to create project BigQuery and generate key file
